### PR TITLE
Fix MorrisExperimentLHS

### PR DESCRIPTION
**Pull request by sofianehaddad**
Fix MorrisExperimentLHS::generate when interval is not standard uniform ([0,1]^dimension)

### CODE REVIEW

**Review by jschueller — APPROVED** [review]
works for me!

File: lib/src/MorrisExperimentLHS.cxx [hunk 1]
```diff
@@ -158,14 +151,21 @@ Sample MorrisExperimentLHS::generateTrajectory(const UnsignedInteger index) cons
...
+    // still belong to the initial domain
+    const Scalar xAxis(xBase[axis] + delta_[axis] * directions[i]);
+    const Scalar xAxisAlternative(xBase[axis] - delta_[axis] * directions[i]);
+    if ((lowerBound[axis] <= xAxis) && (xAxis <= upperBound[axis]))
+      xBase[axis] = xAxis;
+    else if ((lowerBound[axis] <= xAxisAlternative) && (xAxisAlternative <= upperBound[axis]))
+      xBase[axis] = xAxisAlternative;
     else
-      throw InvalidArgumentException(HERE) << "Seems that could not define a path" ;
+      throw InvalidArgumentException(HERE) << "Traying to define a path but " <<  xAxis << " and " << xAxisAlternative << " do no belong the initial domain" ;
```
Comment (jschueller) [inline lib/src/MorrisExperimentLHS.cxx]: Trying

Comment (sofianehaddad) [reply lib/src/MorrisExperimentLHS.cxx]: Ouch! indeed

Comment (sofianehaddad) [reply lib/src/MorrisExperimentLHS.cxx]: Thanks

**Review by dbarbier — APPROVED** [review]
+1

File: python/src/MorrisExperimentLHS_doc.i.in [hunk 2]
```diff
@@ -3,9 +3,9 @@
 
 Available constructors:
 
-    MorrisExperiment(lhsDesign, N)
+    MorrisExperimentLHS(lhsDesign, N)
 
-    MorrisExperiment(lhsDesign, interval, N)
+    MorrisExperimentLHS(lhsDesign, interval, N)
```
Comment (jschueller) [inline python/src/MorrisExperimentLHS_doc.i.in:8]: I think the MorrisExperiment interface class is not necessary in the first place, and even MorrisExperimentImplementation could be removed as tit doesnt factorize anything.

Comment (sofianehaddad) [reply python/src/MorrisExperimentLHS_doc.i.in:8]: @jschueller yes and no. The implementation class factorizes some attribut & methods, checks... 
It dervies from WeightedExperimentImplementation but I don't know if it is a good idea?
Btw, the two **main** experiment classes (MorrisExperimentGrid and MorrisExperimentLHS) have similarities : only the first point (starting point) selection differs. 
Maybe we could think later about actorizing the classes?

Comment (jschueller) [reply python/src/MorrisExperimentLHS_doc.i.in:8]: Ok I see, so the interface can still be removed. It's totally ok to inherit WeightedExperimentImplementation.

Comment (sofianehaddad) [reply python/src/MorrisExperimentLHS_doc.i.in:8]: ok It will be removed in a different PR

**Comment by sofianehaddad** [comment]
@jschueller ok to merge it?
Indeed since 2ea725e04c1350d486038ff0b37bb29c14758617 MorrisExperimentLHS and MorrisExperimentGrid behave differently and there is no reason to keep MorrisExperiment / MorrisExperimentImplementation

I will make  MorrisExperimentLHS and MorrisExperimentGrid inherit WeightedExperimentImplementation next week

**Comment by jschueller** [comment]
Yes, possibly it could have also an impact on the Morris class as the different ctor just have the same signatures of MorrisExperimentGrid/MorrisExperimentLHS, maybe it could be just an experiment ?

**Pull request merged**